api/Trace/SpanOptions.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Trace;
+
+/**
+ * The SpanOptions implementation is intended to be tightly coupled with the Span and Tracer implementations.
+ * Hopefully the Span object can implement SpanOptions so that the toSpan() call is almost a no-op, but prevents
+ * the rest of the SpanOptions API from being used after toSpan() is called.
+ */
+interface SpanOptions
```

Comment:
If I understand it right, this is some kind of Span Builder?
I assumed that if one would want to create a span, then he should pass all parameters in the Span constructor.

Reply:
Yes, it's some sort of builder. However, you can't use a span's constructor; it's forbidden in the spec (citation needed).